src/main/webapp/app/course/competencies/competency-management/competency-management.component.ts

```diff
@@ -69,12 +70,10 @@ export class CompetencyManagementComponent implements OnInit, OnDestroy {
     private readonly featureToggleService: FeatureToggleService = inject(FeatureToggleService);
 
     ngOnInit(): void {
-        this.activatedRoute.parent!.params.subscribe((params) => {
-            this.courseId = params['courseId'];
-            if (this.courseId) {
-                this.loadData();
-                this.loadIrisEnabled();
-            }
+        this.activatedRoute.parent!.params.subscribe(async (params) => {
+            this.courseId = Number(params['courseId']);
+            await this.loadData();
```

Comment:
I removed the if clause as we are 100% sure that the course Id is defined. It's defined as it is a URL param therefore if it would not be defined we would not visit the page at all.